streams/src/main/java/org/apache/kafka/streams/processor/internals/RecordCollectorImpl.java

```diff
@@ -98,10 +108,17 @@ public void onCompletion(RecordMetadata metadata, Exception exception) {
         }
     }
 
+    private void checkForException() {
+        if (sendException != null) {
+            throw new StreamsException(String.format("%s exception caught when producing", logPrefix), sendException);
+        }
+    }
+
     @Override
     public void flush() {
         log.debug("{} Flushing producer", logPrefix);
         this.producer.flush();
+        checkForException();
```

Comment:
Just for clarification. If we call flush, we now that there are no "dangling callbacks" for `onCompletion()` ?